massa-bootstrap/src/client.rs

```diff
@@ -167,6 +167,8 @@ async fn stream_final_state_and_consensus(
                         last_ops_step: StreamingStep::Started,
                         last_consensus_step: StreamingStep::Started,
                     };
+                    let mut write_final_state = global_bootstrap_state.final_state.write();
+                    write_final_state.reset();
                     return Err(BootstrapError::GeneralError(String::from("Slot too old")));
```

Comment:
as mentionned above, instead of clearing things at every failure point (and probably forgetting to do it at some points, and probably forgetting to clear newly added elements), we should better drop and recreate all objects whenever the bootstrap fails and needs a retry (for any reason !)

Reply:
is this taken into account in https://github.com/massalabs/massa/pull/3698 ? @bilboquet